sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Attributes.scala

```diff
@@ -24,15 +24,19 @@ import org.typelevel.otel4s.Attribute
 import org.typelevel.otel4s.Attribute.KeySelect
 import org.typelevel.otel4s.AttributeKey
 
+import scala.collection.IterableOps
 import scala.collection.SpecificIterableFactory
+import scala.collection.immutable
 import scala.collection.mutable
 
 /** An immutable collection of [[Attribute]]s. It contains only unique keys.
   *
   * @see
   *   [[https://opentelemetry.io/docs/specs/otel/common/#attribute-collections]]
   */
-sealed trait Attributes extends Iterable[Attribute[_]] {
+sealed trait Attributes
+    extends immutable.Iterable[Attribute[_]]
+    with IterableOps[Attribute[_], immutable.Iterable, Attributes] {
```

Comment:
this makes sure that things like `filter` still return an `Attributes` rather than an `Iterable[Attribute[_]]` (in practice a `List`)